### ISSUE TITLE
Make confess backtraces log-level dependent?

### ISSUE THREAD

**Issue by terrorobe** [pgmasters/backrest#89]
I find the backtraces a bit unintuitive since they provide no added value for the user and can "wall-of-text" away the real error message since they dominate the output.

Might be a good move to make them only appear for certain log levels (debug & trace?)

Example:

```
2015-05-29 03:00:02.755 T00   WARN: No full backup exists, differential backup has been changed to full
2015-05-29 03:00:04.145 T00   WARN: aborted backup of same type exists, will be cleaned to remove invalid files and resumed
could not find WAL segment 000000010000006D000000A3 after 600.002 second(s) at /opt/backrest/bin/../lib/BackRest/Archive.pm line 147.
        BackRest::Archive::walFileName('BackRest::Archive=HASH(0x125e798)', 'BackRest::File=HASH(0x125cff8)', '000000010000006D000000A3', 600) called at /opt/backrest/bin/../lib/BackRest/Backup.pm line 789
        BackRest::Backup::backup('/data/postgresql/9.3/main', 1) called at /opt/backrest/bin/pg_backrest.pl line 233
        eval {...} called at /opt/backrest/bin/pg_backrest.pl line 107
 at /opt/backrest/bin/pg_backrest.pl line 285.
2015-05-29 03:10:17.642 T00  ERROR: could not find WAL segment 000000010000006D000000A3 after 600.002 second(s)
2015-05-29 03:10:17.646 T00  ERROR: process terminated on signal or exception, 0 threads stopped
```

**Comment by dwsteele** [comment]
Errors that are assigned a code don't get the "wall of text".  I've been slow to get those assigned though since I'm back-filling them.

I like the idea of only displaying the trace when log level is debug or trace.  That would give me some leeway on assigning codes.

**Comment by dwsteele** [comment]
Fixed in d321ef0.

ASSERTs still dump stack traces to the console and file log in all cases.  ERRORs only dump stack traces to the file when the file log level is DEBUG or TRACE.
